### PR TITLE
Add extensive information about the driver in README

### PR DESCRIPTION
**Pull request by Gor027**
## Pre-review checklist

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- [x] PR description sums up the changes and reasons why they should be introduced.
- [ ] I have enabled appropriate tests in `.github/workflows/build.yml` in `gtest_filter`.
- [ ] I have enabled appropriate tests in `.github/workflows/cassandra.yml` in `gtest_filter`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,169 @@
-# cpp-rust-driver
-Wrapper around ScyllaDB's rust-driver compatible with Datastax cpp-driver (WIP)
+# ScyllaDB Cpp-Rust Driver
+___
+Wrapper around ScyllaDB's rust-driver compatible with Datastax cpp-driver.
+As opposed to C/C++ driver, it works with the Cassandra Query Language v4 (CQL4).
```
Comment (constantfold) [inline README.md]: What do you mean by Cassandra Query Language v4? It looks like C/C++ driver works with `CASS_PROTOCOL_VERSION_V4`.

File: README.md [hunk 2]
```diff
@@ -1,12 +1,169 @@
...
+Now, use `--gtest_filter` to run certain integration tests:
+
+```shell
+./cassandra-integration-tests --scylla --install-dir=[PATH_TO_SCYLLA] --version=3.0.8 --category=CASSANDRA --verbose=ccm --gtest_filter="ClusterTests.*"
+```
+
+##### Note: Tests that pass with ScyllaDB and Cassandra clusters can be found in GitHub Actions [`build.yml`](https://github.com/scylladb/cpp-rust-driver/blob/master/.github/workflows/build.yml) and [`cassandra.yml`](https://github.com/scylladb/cpp-rust-driver/blob/master/.github/workflows/cassandra.yml) workflows.
+
+
+# Features
```
Comment (constantfold) [inline README.md]: This should be moved before "Testing" section.

**Comment by constantfold** [comment]
Please also add a section about limitations. Additionally, we should clean up the GitHub Issues that were opened - closing the ones that are now implemented.

File: README.md (hunk 1 above)
Comment (Gor027) [reply README.md]: This was mentioned in the [C++ driver description](https://github.com/scylladb/cpp-driver#compatibility):
#### This driver works exclusively with the Cassandra Query Language v3 (CQL3) and Cassandra's native protocol.

Perhaps, it is outdated information.

File: README.md (hunk 2 above)
Comment (Gor027) [reply README.md]: Do you mean at the beginning of the `Testing` section, or in the description at the very beginning of the document?

Comment (constantfold) [reply README.md]: I mean that the entire "Features" section should be swapped with the entire "Testing" section, changing the order.

File: README.md [hunk 3]
```diff
@@ -1,12 +1,167 @@
...
+# ScyllaDB Cpp-Rust Driver
+___
+Wrapper around ScyllaDB's rust-driver compatible with Datastax cpp-driver.
+
+#### Note: It is work in progress, bug reports and pull requests are welcome!
+
+# Examples
+___
+
+There are some examples in the [examples](https://github.com/scylladb/cpp-driver/tree/master/examples) directory.
```
Comment (Lorak-mmk) [inline README.md]: We have those examples in cpp-rust-driver repo, why point to the old repository?

File: README.md [hunk 4]
```diff
@@ -1,12 +1,167 @@
...
+  * Assuming `scylla` is installed and built in the release mode <br> ``` ln -s [PATH_TO_TOOLS]/scylla-tools-java [PATH_TO_SCYLLA]/resources/cassandra```
+
+Finally, to build the integration tests:
 
 ```
 mkdir build && cd build
-cmake -DCASS_BUILD_INTEGRATION_TESTS=ON  ..
-make
-./cassandra-integration-tests
+cmake -DCASS_BUILD_INTEGRATION_TESTS=ON  .. & make
```
Comment (Lorak-mmk) [inline README.md]: `&` -> `&&`

**Review by Lorak-mmk — CHANGES_REQUESTED** [review]

**Review by Lorak-mmk — APPROVED** [review]

**Review by Lorak-mmk — COMMENTED** [review]
Actually, the limitations section mentiond by @avelanarius is important - please write it before merging.

**Comment by Gor027** [comment]
Sure, I am working on it. I have updated the issues marking functions that are implemented and functions that are still in progress. I will base the limitations section on those issues unless you have another opinion.

**Comment by Gor027** [comment]
V2:

Added a separate section about the limitations of some features.
More information will be added to this section. For now, it covers the main limitations that the current implementation of the driver's features has.

**Comment by Lorak-mmk** [comment]
I'd like for the limitations section to include a note that the section itself may be incomplete - unless you are sure that the list is fairly complete?

One thing I'm not sure about:
In bindings.rs, the function that checks types is unimplemented: https://github.com/scylladb/cpp-rust-driver/blob/5c009e88c6e3d0021d4318855053b0ef514da42c/scylla-rust-wrapper/src/binding.rs#L53-L56
This I think is used for all binding - and you only noted that we are not checking types for `cass_collection_append_*` - is that correct?

**Comment by Gor027** [comment]
> I'd like for the limitations section to include a note that the section itself may be incomplete - unless you are sure that the list is fairly complete?

Sure, I will add a note that the list is incomplete. For now, I have included the main functions which are not implemented or are implemented with some limitations. Functions that are relatively easy to implement (that do not require some implementation in the Rust driver, or maybe even in ScyllaDB) are just noted as `unimplemented`.

> This I think is used for all binding - and you only noted that we are not checking types for `cass_collection_append_*` - is that correct?

What do you mean by `all binding`? The compatibility of types is being checked while appending or setting values in collections, tuples and UDTs. For binding a value to a prepared statement, the type is not being checked and in case of binding with incompatible type, the execution of the prepared statement should not be successful.

**Comment by Lorak-mmk** [comment]
Are you sure that prepared statements are not checked? I did a quick experiment: https://gist.github.com/Lorak-mmk/dbce7e0484b69000f8523df4ad963531

And for me it prints: 
```
===== Using optimized driver!!! =====
bind 1 id: 0
bind 1 somestring: 0
execute 1: 0
bind 2 id: 0
bind 2 somestring: 16777229
execute 2: 0
DONE!
```
Which would mean that they are checked. After reading abstract_data.hh it also seems that the types should be checked for prepared statement.

**Comment by Gor027** [comment]
> Which would mean that they are checked. After reading abstract_data.hh it also seems that the types should be checked for prepared statement.

Yeah, I guess you are right. The `UserTypeValue` and `Statement` classes extend from `AbstractData` class, so setting value by index or name should also check the type compatibility. Will fix the description.

Edit: The Rust driver, too, does not validate the types passed to queries [here](https://github.com/scylladb/scylla-rust-driver/issues/463), however, we can add the validation of types independently from the Rust driver.

**Comment by Lorak-mmk** [comment]
There is already a placeholder function for that, it's just not implemented yet - always returns true.
See bindings.rs:53-56

**Review by Lorak-mmk — APPROVED** [review]

**Pull request merged**